### DISCUSSION TITLE
Scheduler stops on Job throwing exception

### DISCUSSION THREAD

**Discussion by jordyvaneijk** [quartznet/quartznet#529] [General] [answered]
Hi there,

Short question about the job execution. 

I have a couple of jobs that inherit from my BaseJob

`public abstract class BaseJob<T> : IInterruptableJob {
    protected ILoggingService FLoggingService;
    protected ILog Log = LogManager.GetLogger<T>();
    private Thread currentThread;

    public BaseJob() {
      FLoggingService = ServiceLocator.Current.GetInstance<ILoggingService>();
    }

    public void Execute(IJobExecutionContext context) {

      var thread = new Thread(() => {
        try {
          DoWork(context);
        }
        catch (ThreadAbortException e) { //This needs to be catched because when we abort the thread a ThreadAbortException will be raised
          Log.Info($"{typeof(T).Name} killed by user");
          FLoggingService.Log_ApplicationLog(typeof(T).Name, $"{typeof(T).Name} killed by user","");
          throw new JobExecutionException("Job was killed by user");
        }
        catch (Exception e) {
          Log.Error(typeof(T).Name,e);
          FLoggingService.Log_ApplicationError(typeof(T).Name, e);
          throw new JobExecutionException(e,false);
        }
      });

      thread.Start();
      currentThread = thread;
      currentThread.Join();
    }

    public void Interrupt() {
      currentThread.Abort();
    }

    protected abstract void DoWork(IJobExecutionContext AContext);
}`

I also have a simple JobListener that is logging if a job is started and ended. 
As i understand that if i throw a `JobExecutionException` from within the Execute of my job it can be handled inside the `JobWasExecuted(IJobExecutionContext context, JobExecutionException jobException)`? But all it is doing is stopping the server for that matter.

Can someone shine a light on this? Or tell me how to handle exception on a nice way?

Thanks in advance

**Comment by lahma** [comment, marked as answer]
I'm having a bit of a hard time following the logic. Why are you creating separate threads when Quartz already handles that for you?

Your Execute method should probably be just:

```
public void Execute(IJobExecutionContext context)
{
  try 
  {
    DoWork(context);
  }
  catch (Exception e) 
  {
    Log.Error(typeof(T).Name,e);
    FLoggingService.Log_ApplicationError(typeof(T).Name, e);
    throw new JobExecutionException(e,false);
  }
}
```

The Interrupt should just set a flag that you jobs check.
